sphinx_gallery/gen_gallery.py

```diff
@@ -376,6 +376,12 @@ def call_memory(func):
         if gallery_conf['app'] is not None:  # can be None in testing
             gallery_conf['app'].add_css_file(css + '.css')
 
+    # jupyter magics
+    if not isinstance(gallery_conf['promote_jupyter_magic'], bool):
```

Comment:
We should probably use `_bool_eval`, see: https://github.com/sphinx-gallery/sphinx-gallery/blob/66bc0f2b617ddd5d27258cc577574f16bd72650b/sphinx_gallery/gen_gallery.py#L171

Reply:
You're right, that's a cleaner approach. Fixed!